sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Offset.scala

```diff
@@ -23,4 +23,16 @@ package org.apache.spark.sql.execution.streaming
  * ordering of two [[Offset]] instances.  We do assume that if two offsets are `equal` then no
  * new data has arrived.
  */
-trait Offset extends Serializable {}
+abstract class Offset {
+  def json: String
+}
+
+/** Used when loading */
```

Comment:
Improve the docs to something like "A JSON-serialized representation of an Offset that is used for saving offsets to the offset log"